filebeat/harvester/reader/multiline.go

```diff
@@ -113,40 +115,40 @@ func (mlr *Multiline) Next() (Message, error) {
 }
 
 func (mlr *Multiline) readFirst() (Message, error) {
-	for {
-		message, err := mlr.reader.Next()
-		if err != nil {
-			// no lines buffered -> ignore timeout
-			if err == sigMultilineTimeout {
-				continue
-			}
+	//for {
+	message, err := mlr.reader.Next()
+	if err != nil {
+		// no lines buffered -> ignore timeout
+		//if err == sigMultilineTimeout {
+		//	continue
+		//}
 
-			// pass error to caller (next layer) for handling
-			return message, err
-		}
+		// pass error to caller (next layer) for handling
+		return message, err
+	}
 
-		if message.Bytes == 0 {
-			continue
-		}
+	//if message.Bytes == 0 {
+	//	continue
+	//}
 
-		// Start new multiline event
-		mlr.clearBuffer()
-		mlr.resetBuffer(message)
-		mlr.setState((*Multiline).readNext)
-		return mlr.readNext()
-	}
+	// Start new multiline event
+	mlr.clearBuffer()
+	mlr.resetBuffer(message)
+	mlr.setState((*Multiline).readNext)
+	return mlr.readNext()
+	//}
 }
 
 func (mlr *Multiline) readNext() (Message, error) {
 	for {
 		message, err := mlr.reader.Next()
 		if err != nil {
 			// handle multiline timeout signal
-			if err == sigMultilineTimeout {
+			if err == sigMultilineTimeout || err == io.EOF {
 				// no lines buffered -> ignore timeout
-				if mlr.numLines == 0 {
-					continue
-				}
+				//if mlr.numLines == 0 {
+				//	continue
+				//}
```

Comment:
io.EOF and sigMultilineTimeout should still be treated somewhat different. sigMultilinetimeout should never ever be returned from Multiline read methods, but io.EOF must be forwarded. This code block as is will absorb io.EOF as is?